src/analysis/processing/qgsalgorithmnormalraster.cpp

```diff
@@ -0,0 +1,187 @@
+}
+
+QgsNormalRasterAlgorithm *QgsNormalRasterAlgorithm::createInstance() const
+{
+  return new QgsNormalRasterAlgorithm();
+}
+
+void QgsNormalRasterAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterExtent( QStringLiteral( "EXTENT" ), QObject::tr( "Desired extent" ) ) );
+  addParameter( new QgsProcessingParameterCrs( QStringLiteral( "TARGET_CRS" ), QObject::tr( "Target CRS" ), QStringLiteral( "ProjectCrs" ) ) );
+  addParameter( new QgsProcessingParameterNumber( QStringLiteral( "PIXEL_SIZE" ), QObject::tr( "Pixel size" ),
+                QgsProcessingParameterNumber::Double, 1, false, 0.01 ) );
+
+  QStringList rasterDataTypes = QStringList();
+  rasterDataTypes << QStringLiteral( "Float32" )
+                  << QStringLiteral( "Float64" );
+
+  std::unique_ptr< QgsProcessingParameterDefinition > rasterTypeParameter = qgis::make_unique< QgsProcessingParameterEnum >( QStringLiteral( "OUTPUT_TYPE" ), QObject::tr( "Output raster data type" ),  rasterDataTypes, false, 0, false );
+  rasterTypeParameter->setFlags( QgsProcessingParameterDefinition::FlagAdvanced );
+  addParameter( rasterTypeParameter.release() );
+
+  std::unique_ptr< QgsProcessingParameterNumber > meanParameter = qgis::make_unique< QgsProcessingParameterNumber >( QStringLiteral( "MEAN" ), QStringLiteral( "Mean of normal distribution" ), QgsProcessingParameterNumber::Double, 0, true );
+  meanParameter->setFlags( QgsProcessingParameterDefinition::FlagAdvanced );
+  addParameter( meanParameter.release() );
+
+  std::unique_ptr< QgsProcessingParameterNumber > stdevParameter = qgis::make_unique< QgsProcessingParameterNumber >( QStringLiteral( "STDEV" ), QStringLiteral( "Standard deviation of normal distribution" ), QgsProcessingParameterNumber::Double, 1, true, 0 );
+  stdevParameter->setFlags( QgsProcessingParameterDefinition::FlagAdvanced );
+  addParameter( stdevParameter.release() );
+
+  addParameter( new QgsProcessingParameterRasterDestination( QStringLiteral( "OUTPUT" ), QObject::tr( "Output raster" ) ) );
+}
+
+bool QgsNormalRasterAlgorithm::prepareAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
+{
+  Q_UNUSED( feedback )
+  mCrs = parameterAsCrs( parameters, QStringLiteral( "TARGET_CRS" ), context );
+  mExtent = parameterAsExtent( parameters, QStringLiteral( "EXTENT" ), context, mCrs );
+  mPixelSize = parameterAsDouble( parameters, QStringLiteral( "PIXEL_SIZE" ), context );
+  mTypeId = parameterAsInt( parameters, QStringLiteral( "OUTPUT_TYPE" ), context );
+
+  if ( mTypeId == 0 )
+  {
+    mRasterDataType = Qgis::Float32;
+  }
+  else
+  {
+    mRasterDataType = Qgis::Float64;
+  }
+
+  mMean = parameterAsDouble( parameters, QStringLiteral( "MEAN" ), context );
+  mStdev = parameterAsDouble( parameters, QStringLiteral( "STDEV" ), context );
+
+  mNormalDoubleDistribution = std::normal_distribution<double>( mMean, mStdev );
```

Comment:
hard to see if there's issues using these classes across threads -- I'd play it safe and only create them in processAlgorithm

Reply:
fixed